bliss/cached_dataset.py

```diff
@@ -94,19 +193,28 @@ def __init__(
         cached_data_path: str,
         train_transforms: List,
         nontrain_transforms: List,
+        use_iterable_dataset: bool = True,
     ):
         super().__init__()
 
+        self.splits = splits
         self.batch_size = batch_size
         self.num_workers = num_workers
+        self.cached_data_path = pathlib.Path(cached_data_path)
         self.train_transforms = train_transforms
         self.nontrain_transforms = nontrain_transforms
+        self.use_iterable_dataset = use_iterable_dataset
 
-        file_names = [f for f in os.listdir(cached_data_path) if f.endswith(".pt")]
-        self.file_paths = [os.path.join(cached_data_path, f) for f in file_names]
+        self.file_paths = None
+        self.slices = None
 
-        # parse slices from percentages to indices
-        self.slices = self.parse_slices(splits, len(self.file_paths))
+    def setup(self, stage: str) -> None:
+        if stage == "fit":
```

Comment:
Is it necessary to do different things depending on `stage`?

Reply:
this is because i expect it to load the file path after 'prepare data'

Reply:
of course, if we remove the if statement it can also work well. i add the if statement for clear coding

Reply:
in previous implementation, we didn't test the `stage` (in its original interface, `stage` doesn't have a default value)

Reply:
It seems better not to call `setup` if `stage != "fit"`, rather have the entire function body within a conditional block. (Please lmk if I'm missing something.)

Reply:
do you mean we should write the code like:
```python
if stage == "fit":
    setup(stage)
```